fennel/test_lib/mock_client.py

```diff
@@ -766,6 +780,74 @@ def _run_extractors(
                 )
         return output_df
 
+    def _check_exceptions(
+        self, output, dsschema: DSSchema, extractor_name: str
+    ):
+        output_df = pd.DataFrame(output)
+        output_df.reset_index(inplace=True)
+        exceptions = data_schema_check(dsschema, output_df, extractor_name)
+        if len(exceptions) > 0:
+            raise Exception(
+                f"Extractor `{extractor_name}` returned "
+                f"invalid schema: {exceptions}"
+            )
+
+    def _generated_lookup_extractor(
```

Comment:
its not generating an extractor right? it is computing a lookup extractor? The name got me confused a little